Source/WebCore/dom/Element.cpp

```diff
@@ -3238,6 +3298,26 @@ RefPtr<ShadowRoot> Element::shadowRootForBindings(JSC::JSGlobalObject& lexicalGl
     return nullptr;
 }
 
+RefPtr<Element> Element::deepShadowRootReferenceTargetOrSelf() {
+    return const_cast<Element*>(const_cast<const Element*>(this)->deepShadowRootReferenceTargetOrSelf().get());
+}
+
+RefPtr<const Element> Element::deepShadowRootReferenceTargetOrSelf() const {
+    if (!document().settings().shadowRootReferenceTargetEnabled())
+        return this;
+
+    RefPtr<const Element> element = this;
+
+    ShadowRoot* shadow = shadowRoot();
```

Comment:
WebKit's rule for when to use a smart pointer over a raw pointer is that a smart pointer should be used when "non-trivial" functions are called (which are most things that aren't inlined getter / setter type functions).

https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines#when-calling-a-non-trivial-member-function-on-an-object-hold-a-smart-pointer-to-the-object-on-the-stack

So I think we want:

```cpp
RefPtr shadow = shadowRoot();
```